cypress/page/shopping-cart.page.ts

```diff
@@ -0,0 +1,19 @@
+class ShoppingCartPage {
+    private proceedToCheckOutButton: string;
+    private cartNavigationCheckoutButton: string;
+
+    constructor() {
+        this.proceedToCheckOutButton = "[style*='display: block;'] .button-container > a";
+        this.cartNavigationCheckoutButton = ".cart_navigation span";
+    }
+
+    public proceedToCheckout(): void {
+        cy.waitUntil(() => {
+            return cy.get(this.proceedToCheckOutButton, {timeout: 10000}).should("be.visible");
+            });
```

Comment:
this should not be needed when you move the click on the proceedToCheckOutButton property to the product-list.page.ts file